pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -978,4 +978,7 @@ self: super: {
   # https://github.com/mgajda/json-autotype/issues/15
   json-autotype = doJailbreak super.json-autotype;
 
+  # foo
```

Comment:
foo?

Reply:
Sorry, this is still in progress. Tests for `leksah-server` fail after jailbreaking away version restrictions from `ltk` and thus building it with `haskell-gi-overloading-1.0` instead of `haskell-gi-overloading-0.0`. I'm not sure if the tests fail due to this or if they are generally broken

Maybe someone who does actually speak haskell could help here? ;-)

```
# nix-build -j1 -A leksah .
these derivations will be built:
  /nix/store/12f1ybxsxw67wpgqrqypdipzbfscnj7d-leksah-server-0.16.2.0.drv
  /nix/store/w57kzqq5s0d0dydrwv4rjgg95w5vvlyf-leksah-0.16.2.2.drv
  /nix/store/spxdcdaxx8z61wxv1kypa9pi92km5rmf-ghc-8.0.2-with-packages.drv
  /nix/store/sm42a12ccmcjwny7lgmlip7gf64nfxva-leksah.drv
building path(s) ‘/nix/store/6pmnia7f4kxpi7g5iiklwybyq39b0ig9-leksah-server-0.16.2.0-doc’, ‘/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0’, ‘/nix/store/gxwcsszwhk2hi3880lmn9ph92i5qklpl-leksah-server-0.16.2.0-data’
setupCompilerEnvironmentPhase
Build with /nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2.
unpacking sources
unpacking source archive /nix/store/6p1c4vg2rl9flssf8a4vvx90wzz8qz1h-leksah-server-0.16.2.0.tar.gz
source root is leksah-server-0.16.2.0
setting SOURCE_DATE_EPOCH to timestamp 1502752583 of file leksah-server-0.16.2.0/tests/TestTool.hs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-leksah-server-0.16.2.0.drv-0/package.conf.d -j1 -threaded
[1 of 1] Compiling Main             ( Setup.lhs, /tmp/nix-build-leksah-server-0.16.2.0.drv-0/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --datadir=/nix/store/gxwcsszwhk2hi3880lmn9ph92i5qklpl-leksah-server-0.16.2.0-data/share/ghc-8.0.2 --docdir=/nix/store/6pmnia7f4kxpi7g5iiklwybyq39b0ig9-leksah-server-0.16.2.0-doc/share/doc --with-gcc=gcc --package-db=/tmp/nix-build-leksah-server-0.16.2.0.drv-0/package.conf.d --ghc-option=-optl=-Wl,-rpath=/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/lib/ghc-8.0.2/leksah-server-0.16.2.0 --ghc-option=-j1 --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-library-vanilla --enable-executable-dynamic --enable-tests --ghc-option=-split-sections --extra-include-dirs=/nix/store/vx1qc0mfipfsc02p0s466m8j5jmwmkwj-gobject-introspection-1.52.1-dev/include --extra-lib-dirs=/nix/store/vx1qc0mfipfsc02p0s466m8j5jmwmkwj-gobject-introspection-1.52.1-dev/lib --extra-include-dirs=/nix/store/0phm8p3mf0rapxsxcy6v7q298y6a7ax8-libffi-3.2.1-dev/include --extra-lib-dirs=/nix/store/0phm8p3mf0rapxsxcy6v7q298y6a7ax8-libffi-3.2.1-dev/lib --extra-lib-dirs=/nix/store/hhcby2ncg7s75bm8f87j4i7wjzx20nvw-libffi-3.2.1/lib --extra-include-dirs=/nix/store/3jxq33qrlfpjdx17c8d62lmcxfbf2yxm-glib-2.54.0-dev/include --extra-lib-dirs=/nix/store/3jxq33qrlfpjdx17c8d62lmcxfbf2yxm-glib-2.54.0-dev/lib --extra-include-dirs=/nix/store/q8k7r5zbf03ydrpm7imfzds5b2nmdf0a-zlib-1.2.11-dev/include --extra-lib-dirs=/nix/store/q8k7r5zbf03ydrpm7imfzds5b2nmdf0a-zlib-1.2.11-dev/lib --extra-lib-dirs=/nix/store/i8wz3m067dzbs5x2glhxvcg7lvds1942-zlib-1.2.11/lib --extra-include-dirs=/nix/store/d15b3lh1c4452m6w66sjgrw21aps2z8b-glibc-iconv-2.25-49/include --extra-lib-dirs=/nix/store/dq0sfiqgdyxy4z4m1cnplv3ay18kljav-glib-2.54.0/lib --extra-lib-dirs=/nix/store/al907j2gpc7nwlnv2h5w372gx88ipmgb-gobject-introspection-1.52.1/lib
Configuring leksah-server-0.16.2.0...
Flags chosen: network-uri=True, threaded=True
Dependency Cabal >=1.10.2.0 && <2.1: using Cabal-1.24.2.0
Dependency HTTP >=4000.1.2 && <4000.4: using HTTP-4000.3.7
Dependency HUnit >=1.2 && <1.7: using HUnit-1.5.0.0
Dependency aeson >=0.11.2.0 && <1.3: using aeson-1.1.2.0
Dependency attoparsec >=0.10.0.3 && <0.14: using attoparsec-0.13.1.0
Dependency base >=4.0.0.0 && <4.11: using base-4.9.1.0
Dependency base-compat >=0.9.0 && <0.10: using base-compat-0.9.3
Dependency binary >=0.5.0.0 && <0.9: using binary-0.8.3.0
Dependency binary-shared ==0.8.*: using binary-shared-0.8.3
Dependency bytestring >=0.9.0.1 && <0.11: using bytestring-0.10.8.1
Dependency conduit >=1.0.8 && <1.3: using conduit-1.2.12
Dependency conduit-extra >=1.0.0.1 && <1.2: using conduit-extra-1.1.17
Dependency containers >=0.2.0.0 && <0.6: using containers-0.5.7.1
Dependency deepseq >=1.1 && <1.5: using deepseq-1.4.2.0
Dependency directory >=1.0.0.2 && <1.4: using directory-1.3.0.0
Dependency doctest >=0.11.0 && <0.12: using doctest-0.11.4
Dependency executable-path >=0.0.3 && <0.1: using executable-path-0.0.3.1
Dependency filepath >=1.1.0.1 && <1.5: using filepath-1.4.1.1
Dependency ghc >=7.0.4 && <8.3: using ghc-8.0.2
Dependency ghc-boot >=8.0.1 && <8.1: using ghc-boot-8.0.2
Dependency haddock-api >=2.17.2 && <2.18: using haddock-api-2.17.4
Dependency haddock-library >=1.4.1 && <1.5: using haddock-library-1.4.3
Dependency hslogger >=1.0.7 && <1.3: using hslogger-1.2.10
Dependency leksah-server -any: using leksah-server-0.16.2.0
Dependency ltk >=0.16.0.0 && <0.17: using ltk-0.16.1.0
Dependency network ==2.6.*: using network-2.6.3.2
Dependency network-uri ==2.6.*: using network-uri-2.6.1.0
Dependency parsec >=3.0.0.0 && <3.2: using parsec-3.1.11
Dependency pretty >=1.0.1.0 && <1.2: using pretty-1.1.3.3
Dependency process >=1.1 && <1.7: using process-1.4.3.0
Dependency resourcet -any: using resourcet-1.1.9
Dependency strict >=0.3.2 && <0.4: using strict-0.3.2
Dependency text >=0.11.3.1 && <1.3: using text-1.2.2.2
Dependency time >=1.1 && <1.9: using time-1.6.0.1
Dependency transformers >=0.2.2.0 && <0.6: using transformers-0.5.2.0
Dependency unix >=2.3.1.0 && <2.8: using unix-2.7.2.1
Using Cabal-1.24.2.0 compiled by ghc-8.0
Using compiler: ghc-8.0.2
Using install prefix:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0
Binaries installed in:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/bin
Libraries installed in:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/lib/ghc-8.0.2/leksah-server-0.16.2.0
Dynamic libraries installed in:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/lib/ghc-8.0.2/x86_64-linux-ghc-8.0.2
Private binaries installed in:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/libexec
Data files installed in:
/nix/store/gxwcsszwhk2hi3880lmn9ph92i5qklpl-leksah-server-0.16.2.0-data/share/ghc-8.0.2/x86_64-linux-ghc-8.0.2/leksah-server-0.16.2.0
Documentation installed in:
/nix/store/6pmnia7f4kxpi7g5iiklwybyq39b0ig9-leksah-server-0.16.2.0-doc/share/doc
Configuration files installed in:
/nix/store/cmir164p6f3cqzk4ylfiwqqclz8l9cbd-leksah-server-0.16.2.0/etc
No alex found
Using ar found on system at:
/nix/store/z470j6lybdsy4ql972k392490bprhd2g-binutils-2.28.1/bin/ar
No c2hs found
No cpphs found
Using gcc version 6.4.0 given by user at:
/nix/store/kd7vb1afwvm2k65n9qjdw767w6lnwsrd-gcc-wrapper-6.4.0/bin/gcc
Using ghc version 8.0.2 found on system at:
/nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2/bin/ghc
Using ghc-pkg version 8.0.2 found on system at:
/nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.17.3 found on system at:
/nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2/bin/hpc
Using hsc2hs version 0.68.1 found on system at:
/nix/store/m74s3ylqgnpycnhaj1hxqax3xpx9l9z9-ghc-8.0.2/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/s98kd37460pd00iljsf7xkfqxx5nmzzm-hscolour-1.24.1/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/kd7vb1afwvm2k65n9qjdw767w6lnwsrd-gcc-wrapper-6.4.0/bin/ld
No lhc found
No lhc-pkg found
No pkg-config found
Using strip version 2.28 found on system at:
/nix/store/z470j6lybdsy4ql972k392490bprhd2g-binutils-2.28.1/bin/strip
Using tar found on system at:
/nix/store/kpnj0h0340wd0i86q0523527ikbz62ll-gnutar-1.29/bin/tar
No uhc found
building
Building leksah-server-0.16.2.0...
Preprocessing library leksah-server-0.16.2.0...
[ 1 of 18] Compiling Paths_leksah_server ( dist/build/autogen/Paths_leksah_server.hs, dist/build/Paths_leksah_server.o )
[ 2 of 18] Compiling IDE.Utils.Utils  ( src/IDE/Utils/Utils.hs, dist/build/IDE/Utils/Utils.o )
[ 3 of 18] Compiling IDE.Utils.Tool   ( src/IDE/Utils/Tool.hs, dist/build/IDE/Utils/Tool.o )
[ 4 of 18] Compiling IDE.Utils.VersionUtils ( src/IDE/Utils/VersionUtils.hs, dist/build/IDE/Utils/VersionUtils.o )
[ 5 of 18] Compiling IDE.Utils.Server ( src/IDE/Utils/Server.hs, dist/build/IDE/Utils/Server.o )
[ 6 of 18] Compiling IDE.Utils.CabalProject ( src/IDE/Utils/CabalProject.hs, dist/build/IDE/Utils/CabalProject.o )
[ 7 of 18] Compiling IDE.Utils.CabalPlan ( src/IDE/Utils/CabalPlan.hs, dist/build/IDE/Utils/CabalPlan.o )
[ 8 of 18] Compiling IDE.Core.CTypes  ( src/IDE/Core/CTypes.hs, dist/build/IDE/Core/CTypes.o )
[ 9 of 18] Compiling IDE.Core.Serializable ( src/IDE/Core/Serializable.hs, dist/build/IDE/Core/Serializable.o )
[10 of 18] Compiling IDE.StrippedPrefs ( src/IDE/StrippedPrefs.hs, dist/build/IDE/StrippedPrefs.o )
[11 of 18] Compiling IDE.Utils.FileUtils ( src/IDE/Utils/FileUtils.hs, dist/build/IDE/Utils/FileUtils.o )
[12 of 18] Compiling IDE.Metainfo.SourceDB ( src/IDE/Metainfo/SourceDB.hs, dist/build/IDE/Metainfo/SourceDB.o )
[13 of 18] Compiling IDE.Utils.GHCUtils ( src/IDE/Utils/GHCUtils.hs, dist/build/IDE/Utils/GHCUtils.o )
[14 of 18] Compiling IDE.HeaderParser ( src/IDE/HeaderParser.hs, dist/build/IDE/HeaderParser.o )
[15 of 18] Compiling IDE.Metainfo.InterfaceCollector ( src/IDE/Metainfo/InterfaceCollector.hs, dist/build/IDE/Metainfo/InterfaceCollector.o )

src/IDE/Metainfo/InterfaceCollector.hs:27:1-45: warning: [-Wdodgy-imports]
    Module ‘Module’ does not export ‘PackageKey’
[16 of 18] Compiling IDE.Metainfo.WorkspaceCollector ( src/IDE/Metainfo/WorkspaceCollector.hs, dist/build/IDE/Metainfo/WorkspaceCollector.o )

src/IDE/Metainfo/WorkspaceCollector.hs:43:1-60: warning: [-Wdodgy-imports]
    Module ‘GHC’ does not export ‘PackageKey’
[17 of 18] Compiling IDE.Metainfo.SourceCollectorH ( src/IDE/Metainfo/SourceCollectorH.hs, dist/build/IDE/Metainfo/SourceCollectorH.o )
[18 of 18] Compiling IDE.Metainfo.PackageCollector ( src/IDE/Metainfo/PackageCollector.hs, dist/build/IDE/Metainfo/PackageCollector.o )
Preprocessing executable 'leksah-server' for leksah-server-0.16.2.0...
[1 of 1] Compiling Main             ( main/Collector.hs, dist/build/leksah-server/leksah-server-tmp/Main.dyn_o )
Linking dist/build/leksah-server/leksah-server ...
Preprocessing executable 'leksahecho' for leksah-server-0.16.2.0...
[1 of 1] Compiling Main             ( main/LeksahEcho.hs, dist/build/leksahecho/leksahecho-tmp/Main.dyn_o )
Linking dist/build/leksahecho/leksahecho ...
Preprocessing executable 'leksahtrue' for leksah-server-0.16.2.0...
[1 of 1] Compiling Main             ( main/LeksahTrue.hs, dist/build/leksahtrue/leksahtrue-tmp/Main.dyn_o )
Linking dist/build/leksahtrue/leksahtrue ...
Preprocessing test suite 'test-tool' for leksah-server-0.16.2.0...
[1 of 1] Compiling Main             ( tests/TestTool.hs, dist/build/test-tool/test-tool-tmp/Main.dyn_o )

tests/TestTool.hs:(24,1)-(26,17): warning: [-Wunused-imports]
    The import of ‘toolProcess’
    from module ‘IDE.Utils.Tool’ is redundant

tests/TestTool.hs:27:1-67: warning: [-Wunused-imports]
    The import of ‘System.Process’ is redundant
      except perhaps to import instances from ‘System.Process’
    To import instances alone, use: import System.Process()

tests/TestTool.hs:(28,1)-(30,50): warning: [-Wunused-imports]
    The import of ‘@=?, Test, assertBool, runTestTT’
    from module ‘Test.HUnit’ is redundant

tests/TestTool.hs:(34,1)-(35,61): warning: [-Wunused-imports]
    The import of ‘forkIO, threadDelay’
    from module ‘Control.Concurrent’ is redundant

tests/TestTool.hs:37:1-28: warning: [-Wunused-imports]
    The import of ‘Control.Monad’ is redundant
      except perhaps to import instances from ‘Control.Monad’
    To import instances alone, use: import Control.Monad()

tests/TestTool.hs:(38,1)-(39,53): warning: [-Wunused-imports]
    The import of ‘System.Log.Logger’ is redundant
      except perhaps to import instances from ‘System.Log.Logger’
    To import instances alone, use: import System.Log.Logger()

tests/TestTool.hs:40:1-32: warning: [-Wunused-imports]
    The import of ‘System.Log’ is redundant
      except perhaps to import instances from ‘System.Log’
    To import instances alone, use: import System.Log()

tests/TestTool.hs:46:1-5: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      check :: [ToolOutput] -> [ToolOutput] -> IO ()

tests/TestTool.hs:56:1-8: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      runTests :: GHC.MVar.MVar (Maybe (IO a)) -> IO ()

tests/TestTool.hs:61:18-21: warning: [-Wname-shadowing]
    This binding for ‘test’ shadows the existing binding
      imported from ‘Test.HUnit’ at tests/TestTool.hs:30:28-39
      (and originally defined in ‘Test.HUnit.Base’)

tests/TestTool.hs:62:17-20: warning: [-Wunused-do-bind]
    A do-notation statement discarded a result of type ‘a’
    Suppress this warning by saying ‘_ <- test’

tests/TestTool.hs:66:1-8: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      sendTest :: Control.Monad.IO.Class.MonadIO m =>
                  GHC.MVar.MVar (Maybe a) -> a -> m ()

tests/TestTool.hs:66:19-22: warning: [-Wname-shadowing]
    This binding for ‘test’ shadows the existing binding
      imported from ‘Test.HUnit’ at tests/TestTool.hs:30:28-39
      (and originally defined in ‘Test.HUnit.Base’)

tests/TestTool.hs:69:1-11: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      doneTesting :: Control.Monad.IO.Class.MonadIO m =>
                     GHC.MVar.MVar (Maybe a) -> m ()

tests/TestTool.hs:72:1-5: warning: [-Wmissing-signatures]
    Top-level binding with no type signature: tests :: String -> Test

tests/TestTool.hs:236:30-37: warning: [-Wname-shadowing]
    This binding for ‘failures’ shadows the existing binding
      imported from ‘Test.HUnit’ at tests/TestTool.hs:29:40-49
      (and originally defined in ‘Test.HUnit.Base’)
Linking dist/build/test-tool/test-tool ...
running tests
Running 1 test suites...
Test suite test-tool: RUNNING...
### Failure in src/IDE/Utils/CabalPlan.hs:37: expression `:{
  eitherDecodeStrict' (BS.pack
    "{\
    \   \"cabal-lib-version\" : \"1.24.0.0\",\
    \   \"cabal-version\" : \"1.24.0.0\",\
    \   \"install-plan\" : [\
    \      {\
    \         \"type\" : \"pre-existing\",\
    \         \"components\" : {\
    \            \"lib\" : {\
    \               \"depends\" : [\
    \                  \"array-0.5.1.1\",\
    \                  \"base-4.9.0.0\"\
    \               ]\
    \            }\
    \         },\
    \         \"id\" : \"Cabal-1.24.0.0\"\
    \      },\
    \      {\
    \         \"id\" : \"QuickCheck-2.9.1-ec9a1c39266d75ed2c3314f6e846a8f11853eff43fc45db79c7256d9bfd94602\",\
    \         \"components\" : {\
    \            \"lib\" : {\
    \               \"depends\" : [\
    \                  \"base-4.9.0.0\",\
    \                  \"containers-0.5.7.1\",\
    \                  \"random-1.1-fe6ccf72ebd63a2d68570bb45b42bd08df5570c6151cb9af54907d40ef9af454\"\
    \               ]\
    \            }\
    \         },\
    \         \"flags\" : {\
    \            \"templatehaskell\" : true\
    \         },\
    \         \"type\" : \"configured\"\
    \      }\
    \   ]\
    \}") :: Either String PlanJson
:}'
expected: Right (PlanJson {pjPlan = [PlanItem {piId = "Cabal-1.24.0.0", piType = "pre-existing", piComps = [(ComponentLib,fromList ["array-0.5.1.1","base-4.9.0.0"])]},PlanItem {piId = "QuickCheck-2.9.1-ec9a1c39266d75ed2c3314f6e846a8f11853eff43fc45db79c7256d9bfd94602", piType = "configured", piComps = [(ComponentLib,fromList ["base-4.9.0.0","containers-0.5.7.1","random-1.1-fe6ccf72ebd63a2d68570bb45b42bd08df5570c6151cb9af54907d40ef9af454"])]}]})
 but got: Right (PlanJson {pjPlan = [PlanItem {piId = "Cabal-1.24.0.0", piType = "pre-existing", piComps = [(ComponentLib,fromList ["array-0.5.1.1","base-4.9.0.0"])]},PlanItem {piId = "QuickCheck-2.9.1-ec9a1c39266d75ed2c3314f6e846a8f11853eff43fc45db79c7256d9bfd94602", piType = "configured", piComps = [(ComponentLib,fromList ["base-4.9.0.0","containers-0.5.7.1","random-1.1-fe6ccf72ebd63a2d68570bb45b42bd08df5570c6151cb9af54907d40ef9af454"])]}], pjCabalVersion = "1.24.0.0", pjCabalLibVersion = "1.24.0.0", pjCompilerId = Nothing, pjOS = Nothing, pjArch = Nothing})
Examples: 6  Tried: 6  Errors: 0  Failures: 1
Test suite test-tool: FAIL
Test suite logged to: dist/test/leksah-server-0.16.2.0-test-tool.log
0 of 1 test suites (0 of 1 test cases) passed.
builder for ‘/nix/store/12f1ybxsxw67wpgqrqypdipzbfscnj7d-leksah-server-0.16.2.0.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/spxdcdaxx8z61wxv1kypa9pi92km5rmf-ghc-8.0.2-with-packages.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/sm42a12ccmcjwny7lgmlip7gf64nfxva-leksah.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/sm42a12ccmcjwny7lgmlip7gf64nfxva-leksah.drv’ failed

```

Reply:
This looks like a bug in the test suite. It's receiving a more elaborate install plan from Cabal than it expected, i.e. there are additional fields added into it. Maybe our version of Leksah doesn't work with Cabal 1.24.0.0? I'd ask upstream about this issue.

Reply:
@peti Thanks!

Reply:
It seems `leksah`s tests where disabled, as they are currently broken: https://github.com/leksah/leksah/commit/2fb3da75e32a79815dcfb2d5da16ac6d15caa351

I will update the PR to disable tests, too.